frame/bridge/ecdsa-authority/src/lib.rs

```diff
@@ -0,0 +1,427 @@
+
+			let (authorities_count, pre) = <Authorities<T>>::try_mutate(|authorities| {
+				let i =
+					authorities.iter().position(|a| a == &old).ok_or(<Error<T>>::NotAuthority)?;
+
+				<PreviousAuthorities<T>>::put(&*authorities);
+				authorities[i] = new;
+
+				Ok::<_, DispatchError>((
+					authorities.len() as u32,
+					if i == 0 { AUTHORITY_SENTINEL } else { authorities[i - 1] },
+				))
+			})?;
+
+			Self::on_authorities_change(Method::SwapMembers { pre, old, new }, authorities_count);
+
+			Ok(())
+		}
+
+		#[pallet::weight(10_000_000)]
+		#[frame_support::transactional]
+		pub fn submit_authorities_change_signature(
+			origin: OriginFor<T>,
+			address: Address,
+			signature: Signature,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let previous_authorities = Self::ensure_previous_authority(&address)?;
+			let mut authorities_change_to_sign =
+				<AuthoritiesChangeToSign<T>>::take().ok_or(<Error<T>>::NoAuthoritiesChange)?;
+			let (message, collected) = &mut authorities_change_to_sign;
+
+			ensure!(
+				Sign::verify_signature(&signature, message, &address),
+				<Error<T>>::BadSignature
+			);
+
+			collected.try_push((address, signature)).map_err(|_| <Error<T>>::TooManyAuthorities)?;
+
+			if Perbill::from_rational(collected.len() as u32, previous_authorities.len() as u32)
+				>= T::SignThreshold::get()
+			{
+				let authorities_change_to_sign =
+					(authorities_change_to_sign.0, authorities_change_to_sign.1.to_vec());
+
+				Self::deposit_event(<Event<T>>::CollectedEnoughAuthoritiesChangeSignatures(
+					authorities_change_to_sign,
+				));
+			} else {
+				<AuthoritiesChangeToSign<T>>::put(authorities_change_to_sign);
+			}
+
+			Ok(())
+		}
+
+		#[pallet::weight(10_000_000)]
+		#[frame_support::transactional]
+		pub fn submit_new_message_root_signature(
+			origin: OriginFor<T>,
+			address: Address,
+			signature: Signature,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let authorities = Self::ensure_authority(&address)?;
+			let mut new_message_root_to_sign =
+				<NewMessageRootToSign<T>>::take().ok_or(<Error<T>>::NoNewMessageRoot)?;
+			let (message, collected) = &mut new_message_root_to_sign;
+
+			ensure!(
+				Sign::verify_signature(&signature, message, &address),
+				<Error<T>>::BadSignature
+			);
+
+			collected.try_push((address, signature)).map_err(|_| <Error<T>>::TooManyAuthorities)?;
+
+			if Perbill::from_rational(collected.len() as u32, authorities.len() as u32)
+				>= T::SignThreshold::get()
+			{
+				let new_message_root_to_sign =
+					(new_message_root_to_sign.0, new_message_root_to_sign.1.to_vec());
+
+				Self::deposit_event(<Event<T>>::CollectedEnoughNewMessageRootSignatures(
+					new_message_root_to_sign,
+				));
+			} else {
+				<NewMessageRootToSign<T>>::put(new_message_root_to_sign);
+			}
+
+			Ok(())
+		}
+	}
+	impl<T: Config> Pallet<T> {
+		fn network_id() -> NetworkId {
+			network_ids::convert(T::Version::get().spec_name.as_ref())
```

Comment:
Sometimes I'm confused with this naming and Ethereum network id.

But this may not be the scope of this PR